### ISSUE TITLE
Update background color of selected Tree View item to meet contrast ratio requirement

### ISSUE THREAD

**Issue by YuliKl** [microsoft/microsoft-ui-xaml#2926]
**Describe the bug**

Selected TreeViewItems do not currently meet the 3:1 contrast ratio requirement relative to unselected items, as specified in [MAS 1.4.11 - Non-text Contrast](https://microsoft.sharepoint.com/:w:/r/teams/msenable/_layouts/15/Doc.aspx?sourcedoc=%7B6CA3BF14-2635-434A-832C-28DD9A18B9FC%7D&file=MAS%201.4.11%20%E2%80%93%20Non-text%20Contrast.docx&action=default&mobileredirect=true&cid=e2ed3f7b-f231-4f75-a423-80c7558563da).

Light theme ratio is 1.5:1
![image](https://user-images.githubusercontent.com/13142665/87610969-b40a9d80-c6bb-11ea-8fe2-5a7f754650ae.png)

Dark theme ratio is 1.3:1
![image](https://user-images.githubusercontent.com/13142665/87611023-d56b8980-c6bb-11ea-8a96-349098f3d925.png)

**Additional context**

This issue is closely related to #2908 but requires different theme resources to be updated.

**Proposed fix**
Create new resources
```xaml
<RevealBackgroundBrush x:Key="SystemControlHighlightAccent1RevealBackgroundBrush" TargetTheme="Dark" Color="{ThemeResource SystemAccentColorDark1}" FallbackColor="{ThemeResource SystemAccentColorDark1}" />
<RevealBackgroundBrush x:Key="SystemControlHighlightAccent1RevealBackgroundBrush" TargetTheme="Light" Color="{ThemeResource SystemAccentColorLight1}" FallbackColor="{ThemeResource SystemAccentColorLight1}" />
```
Update `TreeViewItemBackgroundSelected` to use the new `SystemControlHighlightAccent1RevealBackgroundBrush`.

**Comment by Felix-Dev** [comment]
Is there a reason the TreeView uses different brushes than the ListView here? Wouldn't it make sense here to use the same brushes for the selected ListViewItems and selected TreeViewItems?

Also: The selection brushes used for the TreeView rcurrently have [an issue](https://github.com/microsoft/microsoft-ui-xaml/issues/1189#issuecomment-658115382) where they don't pick up in-app overrides to the `SystemAccentColor` resource. This works just fine with the brushes the ListView uses so if we can create a single set of brushes to use here for both TreeView and ListView - we should follow the example set by the ListView here.

**Comment by mdtauk** [comment]
@Felix-Dev  The TreeView would need it's own styles so they can be overridden, but those brushes could call upon the same core colour resource as the lists.

But that current colour may be changing, as there are concerns about the contrast ratios at present.

**Comment by Felix-Dev** [comment]
@mdtauk 
> The TreeView would need it's own styles so they can be overridden, but those brushes could call upon the same core colour resource as the lists.

This is exactly what I had in mind. We would still have the various ListViewItemBackroundSelected* and TreeViewItemBackroundSelected* resources (so we can individually style them), but they would both reference then same base brush. So, instead of 
```xml
<StaticResource x:Key="ListViewItemBackgroundSelected" ResourceKey="SystemControlHighlightListAccentMediumLowBrush" />
<StaticResource x:Key="TreeViewItemBackgroundSelected" ResourceKey="SystemControlHighlightAccent1RevealBackgroundBrush" />
```
we would end up with
```xml
<StaticResource x:Key="ListViewItemBackgroundSelected" ResourceKey="SystemControlHighlightListAccentMediumLowBrush" />
<StaticResource x:Key="TreeViewItemBackgroundSelected" ResourceKey="SystemControlHighlightListAccentMediumLowBrush"/>
```
with the added bonus that the TreeViewItem selection brushes now would also pickup SystemAccentColor overrides without having to wait for WinUI 3.

**Comment by Felix-Dev** [comment]
@YuliKl In the related issue #2908 you wrote
> Yes, that's the approach used by TreeView, so I've listed the new colors in #2926. I'm not seriously considering switching ListView's colors to these variants to minimize code churn.

Is there a chance we could switch the TreeView selected item brushes to the brushes usedby the ListView then? We would get some consistent coloring then and also solve the issue reported in #1189? Or should we just update them with your proposed `SystemControlHighlightAccent1RevealBackgroundBrush` theme resource as originally envisioned?

**Comment by dpaulino** [comment]
any chance the work for this issue can also encompass a fix for this? https://github.com/microsoft/microsoft-ui-xaml/issues/1189

**Comment by chingucoding** [comment]
I think that depends on how we fix this. If we just update the opacity values for the existing resources, this won't fix it. But maybe we will update the way the resources in this area work altogether, which could potentially be a breaking change. In that case, it could fix #1189.

**Comment by Felix-Dev** [comment]
Yep, I am asking @YuliKl for this very reason to consider switching the currently used TreeViewItem selected brushes to the one sused by the ListViewItem. If we won't do that, @dpaulino, you can always override those brushes yourself as shown in that issue (while waiting for a fix which would presumably have to wait until WinUI 3).

**Comment by Felix-Dev** [comment]
@YuliKl The ComboBox also uses the same brush for its ComboBoxItemRevealSelected brush as the TreeViewItem currently:
```xml
<StaticResource x:Key="ComboBoxItemRevealBackgroundSelected" ResourceKey="SystemControlHighlightAccent3RevealBackgroundBrush" />
<StaticResource x:Key="ComboBoxItemRevealBackgroundSelectedUnfocused" ResourceKey="SystemControlHighlightAccent3RevealBackgroundBrush" />
<StaticResource x:Key="ComboBoxItemRevealBackgroundSelectedPressed" ResourceKey="SystemControlHighlightAccent3RevealAccent2BackgroundBrush" />
<StaticResource x:Key="ComboBoxItemRevealBackgroundSelectedPointerOver" ResourceKey="SystemControlHighlightAccent2RevealBackgroundBrush" />
```

| Light | Dark |
|---|---|
|![image](https://user-images.githubusercontent.com/1398851/87786448-3b7e1b00-c83a-11ea-8116-f2601f7b6967.png)|![image](https://user-images.githubusercontent.com/1398851/87786469-40db6580-c83a-11ea-91e2-d77097c9d650.png)|

We should thus update those brushes as well (perhaps tracking in a separate issue).

**Comment by YuliKl** [comment]
I'm very hesitant to start fundamentally changing brush values without input from other PM owners. @predavid-zz, @kikisaints, @chigy - any thoughts? 
**TL;DR**: Selected items' background colors aren't meeting 3:1 contrast requirements compared to not selected items. ListViewItems are using partially transparent AccentColor as the selected background. TreeViewItems and ComboBoxItems are using RevealBackgroundBrushes with Color property set to accent color light/dark variants. For consistency between controls and as a workaround for #1189, there's a request to change TreeView and ComboBox to use the same brushes as ListView.
Would changing from using RevealBackgroundBrush to using SolidColorBrush result in visual regressions or other negative impact to these controls?

**Comment by Felix-Dev** [comment]
@YuliKl Do you have an update here about the brushes to use?

**Comment by YuliKl** [comment]
@Felix-Dev I think we have the green light to use the same brushes in TreeView and ComboBox as the updated brushes you added for ListView.

**Comment by Felix-Dev** [comment]
@YuliKl Unfortunately, I think I ran into an issue here when trying to bring over the updated ListView brushes to the TreeView. 

As the TreeViewItem uses reveal highlight, we cannot simply re-use the ListView brushes here. Instead, the idea is to create a new set of *RevealBrushes* which are built the same as the ListView brushes. In other words, if we have a ListViewItem selected rest background brush like 
```xml
<SolidColorBrush x:Key="SystemControlHighlightListAccentMediumLowBrush" Color="{ThemeResource SystemAccentColor}" Opacity="0.75" />
```
we will create an corresponding `RevealBrush` like this:
```xml
<contract5Present:RevealBackgroundBrush x:Key="SystemControlHighlightListAccentMediumLowRevealBackgroundBrush" TargetTheme="Dark" Color="{ThemeResource SystemAccentColor}" Opacity="0.75" FallbackColor="{ThemeResource SystemAccentColorDark1}"/>
```

I thus created a set of new `RevealBrush` equivalents to the used ListViewItem selection brushes (excerpt below shows the brushes defined for the Dark theme):
```xml
<contract5Present:RevealBackgroundBrush x:Key="SystemControlHighlightListAccentVeryHighRevealBackgroundBrush" TargetTheme="Dark" Color="{ThemeResource SystemAccentColor}" Opacity="0.9" FallbackColor="{ThemeResource SystemAccentColor}" />
<contract5Present:RevealBackgroundBrush x:Key="SystemControlHighlightListAccentMediumLowRevealBackgroundBrush" TargetTheme="Dark" Color="{ThemeResource SystemAccentColor}" Opacity="0.75" FallbackColor="{ThemeResource SystemAccentColorDark1}"/>
<contract5Present:RevealBackgroundBrush x:Key="SystemControlHighlightListAccentLowRevealBackgroundBrush" TargetTheme="Dark" Color="{ThemeResource SystemAccentColor}" Opacity="0.6" FallbackColor="{ThemeResource SystemAccentColorDark2}" />
```

These new resources added to WinUI will then be referenced by the relevant TreeView theme resources, like so:
```xml
<StaticResource x:Key="TreeViewItemBackgroundSelected" ResourceKey="SystemControlHighlightListAccentMediumLowRevealBackgroundBrush" />
<StaticResource x:Key="TreeViewItemBackgroundSelectedPointerOver" ResourceKey="SystemControlHighlightListAccentLowRevealBackgroundBrush" />
<StaticResource x:Key="TreeViewItemBackgroundSelectedPressed" ResourceKey="SystemControlHighlightListAccentVeryHighRevealBackgroundBrush" />
```

Unfortunately for us, while the selection background colors are indeed consistent now with the ListView item selection brushes, the reveal highlight effect is missing! See this gif:
![treeview-revealbrush-issue](https://user-images.githubusercontent.com/1398851/90300912-026ab200-de9d-11ea-9054-6da0249b923e.gif)

As you can see, the reveal highlight effect is gone 😢

Note that even the original suggestion for this issue - introducing a new `SystemControlHighlightAccent1RevealBackgroundBrush` theme resource - would suffer from this issue as it is a `RevealBrush` as well.

I have also hit this issue in the past when using WinUI's reveal brushes. See this issue [here](https://github.com/microsoft/microsoft-ui-xaml/issues/2118).

I'm not sure right now if that is something we can fix in WinUI 2 or perhaps need WinUI 3 for that. Perhaps @ranjeshj and @StephenLPeters know more (mention is also meant as a FYI).

Your thoughts here, @YuliKl?

**Comment by YuliKl** [comment]
Closing this issue. TreeView styles and colors are being updated in the near future.
